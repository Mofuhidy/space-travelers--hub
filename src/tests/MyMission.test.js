import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import Mission from '../components/MyMission';

const mockStore = configureMockStore();

test('renders Mission component correctly', () => {
  const mission = {
    missionId: 'mission_123',
    missionName: 'Test Mission',
    description: 'This is a test mission.',
    reserved: true,
  };

  const store = mockStore({
    missions: {
      missions: [],
      activeMissions: [],
      isLoading: false,
    },
  });

  const { asFragment, getByText } = render(
    <Provider store={store}>
      <Mission mission={mission} />
    </Provider>,
  );

  const missionNameElement = getByText(mission.missionName);
  const descriptionElement = getByText(mission.description);
  const activeMemberElement = getByText('Active Member');
  const leaveMissionButton = getByText('Leave Mission');

  expect(asFragment()).toMatchSnapshot();
  expect(missionNameElement).toBeInTheDocument();
  expect(descriptionElement).toBeInTheDocument();
  expect(activeMemberElement).toBeInTheDocument();
  expect(leaveMissionButton).toBeInTheDocument();
});

test('Mission component is in the document', () => {
  const mission = {
    missionId: 'mission_123',
    missionName: 'Test Mission',
    description: 'This is a test mission.',
    reserved: true,
  };

  const store = mockStore({
    missions: {
      missions: [],
      activeMissions: [],
      isLoading: false,
    },
  });

  const { getByText } = render(
    <Provider store={store}>
      <Mission mission={mission} />
    </Provider>,
  );

  const missionNameElement = getByText(mission.missionName);
  const descriptionElement = getByText(mission.description);
  const activeMemberElement = getByText('Active Member');
  const leaveMissionButton = getByText('Leave Mission');

  expect(missionNameElement).toBeInTheDocument();
  expect(descriptionElement).toBeInTheDocument();
  expect(activeMemberElement).toBeInTheDocument();
  expect(leaveMissionButton).toBeInTheDocument();
});
