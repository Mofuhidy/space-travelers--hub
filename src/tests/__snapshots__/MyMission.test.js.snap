// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Mission component correctly 1`] = `
<DocumentFragment>
  Test MissionThis is a test mission.
  <span
    class="mission-reserved"
  >
    Active Member
  </span>
  <button
    class="leave-mission"
    type="button"
  >
    Leave Mission
  </button>
</DocumentFragment>
`;
